lib/net.js

```diff
@@ -325,12 +328,16 @@ Socket.prototype.setNoDelay = function(enable) {
   // backwards compatibility: assume true when `enable` is omitted
   if (this._handle && this._handle.setNoDelay)
     this._handle.setNoDelay(enable === undefined ? true : !!enable);
+  else
+    this._setNoDelay = enable === undefined ? true : !!enable;
 };
 
 
 Socket.prototype.setKeepAlive = function(setting, msecs) {
   if (this._handle && this._handle.setKeepAlive)
     this._handle.setKeepAlive(setting, ~~(msecs / 1000));
+  else
```

Comment:
I thought I commented on this before, but maybe not. By using an `if-else`, don't you take the chance of the `_setKeepAlive` flag having a different value from what is actually being used by the handle (same for `_setNoDelay`)? For example, if you call `setKeepAlive()` after the handle is created, then `_setKeepAlive` will always be `null`. It's not a huge deal since it's internal, but it could be confusing during debugging.

What if you did:

```javascript
this._setKeepAlive = [setting, ~~(msecs / 1000)];

if (this._handle && this._handle.setKeepAlive)
     this._handle.setKeepAlive(setting, ~~(msecs / 1000));
```